types/prosemirror-view/index.d.ts

```diff
@@ -627,6 +627,16 @@ export interface EditorProps<ThisT = unknown, S extends Schema = any> {
    */
   scrollMargin?: number | { top: number, right: number, bottom: number, left: number } | null;
 }
+/**
+ * A mapping of dom events.
+ */
+export type HandleDOMEventsProp<ThisT = unknown, S extends Schema = any> = Partial<
+  {
+    [K in keyof DocumentEventMap]: (this: ThisT, view: EditorView<S>, event: DocumentEventMap[K]) => boolean;
```

Comment:
I like the idea of using `DocumentEventMap` to get more accurate types without hand-coding them ourselves! :+1: